cmp/grouping/GroupingChooserModel.ts

```diff
@@ -187,6 +190,15 @@ export class GroupingChooserModel extends HoistModel {
         this.setFavorites(favorites);
     }
 
+    @action
+    setDimensions(dimensions: Array<DimensionSpec | string>) {
+        throwIf(isEmpty(dimensions), 'Must provide valid dimensions available for selection.');
```

Comment:
I have a use-case where the user defines which fields are available as dimensions, and it is valid to have no dimensions defined. It would make my app code cleaner if this component and model could at least be instantiated/rendered with no dimensions configured when `allowEmpty == true` - avoiding the need to re-create models in cases like this is always a nice cleanup